### PR TITLE
add calcEffectiveAmount into txProcessor

### PR DESCRIPTION
**Pull request by kaiden-araki**
### Add `computeEffectiveAmounts` into `TxProcessor`

`computeEffectiveAmounts` check that the L1Tx data is correct.

- If can not get account for `tx.FromIdx`, return error and `EffectiveAmount` and `EffectiveDepositAmount` will be 0.
- Check the sender has enough balance.
- Check the `tx.FromEthAddr` is the same as `Sender`'s `EthAddr`
- If `txType` is `Exit`, no check.

### CODE REVIEW

File: go_sequencer/txprocessor/txprocessor.go [hunk 1]
```diff
@@ -1242,3 +1243,49 @@ func (txProcessor *TxProcessor) applyExit(coordIdxsMap map[common.TokenID]common
 
 	return exitAccount, false, nil
 }
+
+// computeEffectiveAmounts checks that the L1Tx data is correct
+func (txProcessor *TxProcessor) computeEffectiveAmounts(tx *common.L1Tx) {
+	tx.EffectiveAmount = tx.Amount
+	tx.EffectiveDepositAmount = tx.DepositAmount
+
```
Comment (Shailu-s) [inline go_sequencer/txprocessor/txprocessor.go:1251]: @kaiden-araki Regarding checks for handling the coordinator transactions. We'll need to add checks for the same  - to check if txs is originated by user or by coordinator. 

```
if !tx.UserOrigin {
		// case where the L1Tx is generated by the Coordinator
		tx.EffectiveAmount = big.NewInt(0)
		tx.EffectiveDepositAmount = big.NewInt(0)
		return
	}
	
```

File: go_sequencer/txprocessor/txprocessor.go [hunk 2]
```diff
@@ -1242,3 +1243,49 @@ func (txProcessor *TxProcessor) applyExit(coordIdxsMap map[common.TokenID]common
...
+		return
+	}
+
+	// check that the tx.FromEthAddr is the same than the EthAddress of the
+	// Sender
+	if !bytes.Equal(tx.FromEthAddr.Bytes(), accSender.EthAddr.Bytes()) {
+		log.Debugf("EffectiveAmount = 0: tx.FromEthAddr (%s) must be the same EthAddr of "+
+			"the sender account by the Idx (%s)",
+			tx.FromEthAddr.Hex(), accSender.EthAddr.Hex())
+		tx.EffectiveAmount = big.NewInt(0)
```
Comment (Intizar-T) [inline go_sequencer/txprocessor/txprocessor.go:1283]: I'm wondering if we should `return` here as well 🤔

File: go_sequencer/txprocessor/txprocessor.go [hunk 3]
```diff
@@ -1242,3 +1243,49 @@ func (txProcessor *TxProcessor) applyExit(coordIdxsMap map[common.TokenID]common
...
+
+// computeEffectiveAmounts checks that the L1Tx data is correct
+func (txProcessor *TxProcessor) computeEffectiveAmounts(tx *common.L1Tx) {
+	tx.EffectiveAmount = tx.Amount
+	tx.EffectiveDepositAmount = tx.DepositAmount
+
+	if tx.Type == common.TxTypeCreateAccountDeposit {
+		return
+	}
+
```
Comment (Intizar-T) [inline go_sequencer/txprocessor/txprocessor.go:1255]: the `hermez-node` has a `CreateAccountDepositTransfer` case here. I'm wondering if we don't need it. The check is done by 
```go
if tx.ToIdx >= common.UserThreshold && tx.FromIdx == common.Idx(0) { 
// ...
}
```

Comment (kaiden-araki) [reply go_sequencer/txprocessor/txprocessor.go:1255]: We don't need to check CreateAccountDepositTransfer.

File: go_sequencer/txprocessor/txprocessor.go (hunk 1 above)
Comment (kaiden-araki) [reply go_sequencer/txprocessor/txprocessor.go:1251]: Yeah, let's check coordinator is necessary or not.

File: go_sequencer/txprocessor/txprocessor.go (hunk 2 above)
Comment (kaiden-araki) [reply go_sequencer/txprocessor/txprocessor.go:1283]: Thanks, here we should `return` :sweat_smile:

**Pull request merged**